src/legacy/core_plugins/data/public/expressions/expression_renderer.tsx

```diff
@@ -50,7 +47,8 @@ export const createRenderer = (run: ExpressionRunner): ExpressionRenderer => ({
 
   useEffect(() => {
     if (mountpoint.current) {
-      run(expression, { ...options, element: mountpoint.current }).catch(result => {
+      const handler = loader(mountpoint.current, expression, options);
```

Comment:
The re-creates the loader every time a new expression is passed into this component, right? Shouldn't we hold the loader in a ref? Otherwise cancel isn't called correctly in this branch https://github.com/elastic/kibana/pull/42337/files#diff-1793959dde8c2d221081302bc14dd984R99 for the old loader.